metadata-ingestion/src/datahub_provider/client/dagster_generator.py

```diff
@@ -0,0 +1,278 @@
+import os
+from typing import Dict, List, Optional
+
+from dagster._core.execution.stats import RunStepKeyStatsSnapshot, StepEventStatus
+from dagster._core.snap import JobSnapshot
+from dagster._core.snap.node import OpDefSnap
+from dagster._core.storage.dagster_run import (
+    DagsterRun,
+    DagsterRunStatsSnapshot,
+    DagsterRunStatus,
+)
+
+from datahub.api.entities.datajob import DataFlow, DataJob
+from datahub.api.entities.dataprocess.dataprocess_instance import (
+    DataProcessInstance,
+    InstanceRunResult,
+)
+from datahub.emitter.rest_emitter import DatahubRestEmitter
+from datahub.utilities.urns.data_flow_urn import DataFlowUrn
+from datahub.utilities.urns.data_job_urn import DataJobUrn
+from datahub.utilities.urns.dataset_urn import DatasetUrn
+
+# Default configuration constants
+DEFAULT_DATAHUB_REST_URL = "http://localhost:8080"
+DEFAULT_DATAHUB_ENV = "prod"
+
+# Contants
+ORCHESTRATOR = "dagster"
+
+# Datahub inputs/outputs constant
+DATAHUB_INPUTS = "datahub.inputs"
+DATAHUB_OUTPUTS = "datahub.outputs"
+
+# Job run constant
+JOB_SNAPSHOT_ID = "job_snapshot_id"
+EXECUTION_PLAN_SNAPSHOT_ID = "execution_plan_snapshot_id"
+ROOT_RUN_ID = "root_run_id"
+PARENT_RUN_ID = "parent_run_id"
+HAS_REPOSITORY_LOAD_DATA = "has_repository_load_data"
+TAGS = "tags"
+STEPS_SUCCEEDED = "steps_succeeded"
+STEPS_FAILED = "steps_failed"
+MATERIALIZATIONS = "materializations"
+EXPECTATIONS = "expectations"
+ENQUEUED_TIME = "enqueued_time"
+LAUNCH_TIME = "launch_time"
+START_TIME = "start_time"
+END_TIME = "end_time"
+
+# Op run contants
+STEP_KEY = "step_key"
+ATTEMPTS = "attempts"
+
+# Configuration contants
+DATAHUB_REST_URL: str = (
+    os.environ["DATAHUB_REST_URL"]
+    if "DATAHUB_REST_URL" in os.environ
+    else DEFAULT_DATAHUB_REST_URL
+)
+DATAHUB_ENV: str = (
```

Comment:
add config class in the sensors package and move this into the config class (env, platform_instance and datahub_rest_url)